src/coreclr/gc/gc.cpp

```diff
@@ -6965,8 +6990,13 @@ void gc_heap::gc_thread_function ()
             bool wait_on_time_out_p = gradual_decommit_in_progress_p;
             uint32_t wait_time = DECOMMIT_TIME_STEP_MILLISECONDS;
 #ifdef DYNAMIC_HEAP_COUNT
+#ifdef BACKGROUND_GC
+            bool background_running_p = gc_heap::background_running_p ();
+#else
+            bool background_running_p = false;
+#endif
             // background_running_p can only change from false to true during suspension.
-            if (!gc_heap::background_running_p () && dynamic_heap_count_data.should_change_heap_count)
+            if (!background_running_p && dynamic_heap_count_data.should_change_heap_count)
```

Comment:
you could do 

```cplusplus
if (
#ifdef BACKGROUND_GC
 !gc_heap::background_running_p () && 
#endif
 dynamic_heap_count_data.should_change_heap_count)
```
instead of defining a var called `background_running_p` when BGC code is not even compiled.

Reply:
Would you mind if I simply make background_running_p() defined without BACKGROUND_GC defined and returns false in that case?

There are a few places where things can be simplified that way

Reply:
imagine if we had a version of gc.cpp with code that's only defined when BACKGROUND_GC is defined stripped out, now you end up with background_running_p that's not even a concept that exists. it seems a bit better to avoid that.